### ISSUE TITLE
Strip non alpha numeric characters from the beginning of a comment before validating it

### ISSUE THREAD

**Issue by pixel27** [golang/lint#225]
In addition to documentation I use comments visually break a source file into sections.  This means that many of my comments begin with a row of "****" or "----".  However this runs afoul of the  "comment on export <blah> should be of the form <blah>....".

If those characters could be stripped before the comment text is checked, as the symbols are obviously not documentation that would be helpful.  

Another solution would be (maybe in conjunction with godoc) to have the comment of "private" comments.  Original comments in the code were for developers, then someone got the idea that those comments could also be used for documentation, which is good in general.  However sometimes you still want to provide "internal" documentation that external developers don't need to see.  Often when I want to provide internal documentation for other developers or the future me as to what I was thinking, things to look out for, or the internal workings of a method or object.  

Go appears to have taken over the source comments by saying "Comments on public object and public functions are always available for public consumption."  I believe we still need a way to say hey this comment is for future developers working on this function or object.

**Comment by pixel27** [comment]
For my own personal use I've done the change and placed in pull request #226.

**Comment by dsnet** [comment]
I believe a much easier solution to just put the row of `****` with an extra line break before the next docstring.

For example:
```go
// Bar1 does nothing.
func Bar1() {}

// Bar2 does nothing.
func Bar2() {}

/**********************************/

// Foo1 does nothing.
func Foo1() {}

// Foo2 does nothing.
func Foo2() {}
```

Does this solve your use case? I'm not in favor of stripping alpha-numeric characters from a docstring since it leads to worse issues.

**Comment by pixel27** [comment]
I guess that would work...it just comes down to coding practices.  After using Go a little more i'm just starting to feel like Go's coding guidelines conflict with mine too much to even try to appease lint.

**Comment by frasergraham** [comment]
The guidelines in Effective Go discourage the use of rows of stars or such as separators.

https://golang.org/doc/effective_go.html#commentary

Comments do not need extra formatting such as banners of stars. The generated output may not even be presented in a fixed-width font, so don't depend on spacing for alignment—godoc, like gofmt, takes care of that.

**Comment by dsnet** [comment]
Thanks for pointing to that, @frasergraham. Closing this.
